clippy_utils/src/lib.rs

```diff
@@ -1488,10 +1489,42 @@ pub fn match_function_call<'tcx>(
 /// Checks if `Ty` is normalizable. This function is useful
 /// to avoid crashes on `layout_of`.
 pub fn is_normalizable<'tcx>(cx: &LateContext<'tcx>, param_env: ty::ParamEnv<'tcx>, ty: Ty<'tcx>) -> bool {
-    cx.tcx.infer_ctxt().enter(|infcx| {
+    is_normalizable_helper(cx, param_env, ty, &mut HashMap::new())
+}
+
+fn is_normalizable_helper<'tcx>(
```

Comment:
One last thing: This function should not have to exist, if the `normalize` function in rustc would work as intended. Can you add a `FIXME` linking to the [`normalize`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_trait_selection/infer/at/struct.At.html#method.normalize) documentation, explaining that this helper can be removed in the future, once `normalize` doesn't panic anymore?